src/adapter.rs

```diff
@@ -240,16 +241,18 @@ unsafe fn get_adapter(adapter_addresses_ptr: PIP_ADAPTER_ADDRESSES) -> Result<Ad
 }
 
 unsafe fn socket_address_to_ipaddr(socket_address: &SOCKET_ADDRESS) -> IpAddr {
-    let sockaddr = socket2::SockAddr::from_raw_parts(
-        socket_address.lpSockaddr as *const SOCKADDR,
-        socket_address.iSockaddrLength,
-    );
-
-    // Could be either ipv4 or ipv6
-    sockaddr
-        .as_inet()
-        .map(|s| IpAddr::V4(*s.ip()))
-        .unwrap_or_else(|| IpAddr::V6(*sockaddr.as_inet6().unwrap().ip()))
+    let (_, sockaddr) = socket2::SockAddr::init(|storage, length| {
+        let sockaddr_length = usize::try_from(socket_address.iSockaddrLength).unwrap();
```

Comment:
It would also be possible to use a numeric cast as i think there is a test that this is a valid usize: `bindgen_test_layout_sockaddr()`